src/mngt/ocf_mngt_cache.c

```diff
@@ -350,7 +351,8 @@ static void _ocf_mngt_close_all_uninitialized_cores(
 			continue;
 
 		volume = &(cache->core[i].volume);
-		ocf_volume_close(volume);
+		if (context->cfg.open_cores)
+			ocf_volume_close(volume);
```

Comment:
Since *volume* is used only for closing, maybe you can also initialize it inside this *if* condition?

Reply:
I like it more this way, less lines of code and it's easier to read for me. Compiler will most likely optimize the initialization for us.